frontend/www/js/omegaup/components/coderofthemonth/CoderOfTheMonth.vue

```diff
@@ -18,65 +22,74 @@
           <th></th>
           <th>{{T.codersOfTheMonthCountry}}</th>
           <th>{{T.codersOfTheMonthUser}}</th>
-          <th v-if="showCurrentMonth">{{T.codersOfTheMonthDate}}</th>
+          <th v-if="showCurrentMonth == 1">{{T.codersOfTheMonthDate}}</th>
+          <th v-if="showCurrentMonth == 3">{{T.profileStatisticsNumberOfSolvedProblems}}</th>
+          <th v-if="showCurrentMonth == 3">{{T.rankScore}}</th>
+          <th v-if="showCurrentMonth == 3">{{T.wordsActions}}</th>
         </tr>
       </thead>
       <tbody>
         <tr v-for="coder in visibleCoders">
           <td><img v-bind:src="coder.gravatar_32"></td>
-          <td><img v-bind:src="`/media/flags/${coder.country_id.toLowerCase()}.png`"
-               v-if="coder.country_id != null"></td>
+          <td><omegaup-countryflag v-bind:country="coder.country_id"></omegaup-countryflag></td>
           <td><omegaup-user-username v-bind:classname="coder.classname"
                                  v-bind:linkify="true"
                                  v-bind:username="coder.username"></omegaup-user-username></td>
-          <td v-if="showCurrentMonth">{{coder.date}}</td>
+          <td v-if="showCurrentMonth == 1">{{coder.date}}</td>
+          <td v-if="showCurrentMonth == 3">{{coder.ProblemsSolved}}</td>
+          <td v-if="showCurrentMonth == 3">{{coder.score}}</td>
+          <td v-if="showCurrentMonth == 3"><button class="btn btn-primary"
+                  v-if="canChooseCoder &amp;&amp; !coderIsSelected"
+                  v-on:click=
+                  "onSelectCoder(coder.username)">{{T.coderOfTheMonthChooseAsCoder}}</button></td>
         </tr>
       </tbody>
     </table>
   </div>
 </template>
 
 <script>
-import {API, T} from '../../omegaup.js';
-import UI from '../../ui.js';
+import {T} from '../../omegaup.js';
 import user_Username from '../user/Username.vue';
+import country_Flag from '../CountryFlag.vue';
 
 export default {
-  props: {},
+  props: {
+    codersOfCurrentMonth: Array,
+    codersOfPreviousMonth: Array,
+    candidatesToCoderOfTheMonth: Array,
+    canChooseCoder: Boolean,
+    coderIsSelected: Boolean,
+    isMentor: Boolean,
+  },
   computed: {
     visibleCoders: function() {
-      return this.showCurrentMonth ? this.codersOfCurrentMonth :
-                                     this.codersOfPreviousMonth;
+      switch (this.showCurrentMonth) {
+        case 1:
+          return this.codersOfCurrentMonth;
+        case 2:
+          return this.codersOfPreviousMonth;
+        case 3:
+          return this.candidatesToCoderOfTheMonth;
+        default:
```

Comment:
```suggestion
        case 'codersOfCurrentMonth':
        default:
```

así no tienes que repetir el cuerpo del caso.

Reply:
oh, excelente